lib/ui/discover/discover_page.dart

```diff
@@ -16,33 +94,103 @@ class HomeDiscoverPage extends StatefulWidget {
-  return ListView(
-      padding: const EdgeInsets.only(top: 20.0),
-      children: (articlesList != null) ? (articlesList.map((article) =>
-          DiscoveryListItem().buildListItem(article, getRelatedArticles))).toList() : [Container()]);
-}
+  Widget _buildDiscoverPage(BuildContext context, List<ArticleEntity> articlesList, Function getRelatedArticles, HomeDiscoverPageStyle discoverStyle) {
+    return ListView(
+      children: <Widget>[
+        _buildScreenTitle(discoverStyle),
+        _buildHeaderArticle(articlesList.first, discoverStyle),
+        _buildArticlesList(articlesList, getRelatedArticles, discoverStyle)
+      ],
+    );
+  }
+
+  Widget _buildScreenTitle(HomeDiscoverPageStyle discoverStyle) {
+    return Container(
+      padding: discoverStyle.titlePadding,
+      child: Row(
+        crossAxisAlignment: discoverStyle.leftAlignmentHorizontal,
+        children: <Widget>[
+          Text(Strings.discoverTab, style: discoverStyle.titlePageStyle)
+        ],
+      ),
+    );
+  }
+
+  Widget _buildHeaderArticle(ArticleEntity firstArticle, HomeDiscoverPageStyle discoverStyle) {
+    return Column(
+          children: <Widget>[
+            Container(
+              padding: discoverStyle.headerArticlePadding,
+              child: Column(
+                crossAxisAlignment: discoverStyle.leftAlignmentHorizontal,
+                children: <Widget>[
+                  _buildHeaderArticleImage(firstArticle, discoverStyle),
+                  SizedBox(height: discoverStyle.spaceBetweenHeaderImageAndText),
+                  Text(
+                      firstArticle.title,
+                      maxLines: discoverStyle.headerTitleMaxLines,
+                      style: discoverStyle.titleArticleStyle
+                  ),
+                  SizedBox(height: discoverStyle.spaceBetweenTitleAndSummary),
+                  Text(
+                      firstArticle.summary,
+                      maxLines: discoverStyle.headerSummaryMaxLines,
+                      style: discoverStyle.summaryArticleStyle
+                  )
+                ],
+              ),
+            ),
+            buildListSeparator(discoverStyle)
+          ],
+        );
+  }
+
+  Widget buildListSeparator(HomeDiscoverPageStyle discoverStyle) {
+    return Container(
+        height: discoverStyle.separatorHeight,
+        color: discoverStyle.separatorColor,
+        margin: discoverStyle.separatorIndentation
+    );
+  }
 
+  Widget _buildArticlesList(List<ArticleEntity> articlesList, Function getRelatedArticles, HomeDiscoverPageStyle discoverStyle) {
+    return ListView.builder(
+        shrinkWrap: true,
+        physics: ScrollPhysics(),
+        itemCount: articlesList.length-1,
+        itemBuilder: (BuildContext context, int index) {
+          return DiscoveryListItem().buildListItem(articlesList[index+1], getRelatedArticles, discoverStyle);
```

Comment:
don't like the +1´s and -1´s. I think we should just have a list, but use a different item builder function for the 2 styles.... if index == 0 use heroListItemBuilder else use standardListItemBuilder. This would remove the complexity in the widget building in the above code too.

Reply:
Awesome idea! This solves the problem more elegantly